Content.Shared/CriminalRecords/Systems/SharedCriminalRecordsConsoleSystem.cs

```diff
@@ -1,8 +1,52 @@
+using Content.Shared.IdentityManagement;
+using Content.Shared.IdentityManagement.Components;
+using Content.Shared.Security;
+using Content.Shared.Security.Components;
+
 namespace Content.Shared.CriminalRecords.Systems;
 
-/// <summary>
-/// Nothing is predicted just exists for access.
-/// </summary>
 public abstract class SharedCriminalRecordsConsoleSystem : EntitySystem
 {
+    /// <summary>
+    /// Any entity that has a the name of the record that was just changed as their visible name will get their icon
+    /// updated with the new status, if the record got removed their icon will be removed too.
+    /// </summary>
+    public void UpdateCriminalIdentity(string name, SecurityStatus status)
+    {
+        var query = EntityQueryEnumerator<IdentityComponent>();
+
+        while (query.MoveNext(out var uid, out var identity))
+        {
+            if (!Identity.Name(uid, EntityManager).Equals(name))
+                continue;
+
+            if (status == SecurityStatus.None)
+                RemComp<CriminalRecordComponent>(uid);
```

Comment:
Better find a other solution from adding and removing the component for show a icon, like a null icon